### ISSUE TITLE
Retry doesnt work when started offline

### ISSUE THREAD

**Issue by daleharvey** [pouchdb/pouchdb#4049]

**Comment by daleharvey** [comment]
Hrm, should have been fixed in https://github.com/pouchdb/pouchdb/issues/3768

**Comment by daleharvey** [comment]
ok confirmed this is still a bug, for some reason the checkpointing is failing looking into it now, not sre why previous test isnt catching it

**Comment by nolanlawson** [comment]
I tested manually using the Chrome dev tools to toggle offline and online, and it seemed to work. I may have missed something though.

**Comment by daleharvey** [comment]
So the issue here is that we cant construct a http pouchdb instance while we are offline, it errors out and `isReady` never gets set to true, it doesnt look like we reconstruct the database on retry

**Comment by daleharvey** [comment]
The test doesnt cover that because it overrides the databases methods after construction

**Comment by daleharvey** [comment]
This isnt particularly easy to fix, we cant simply reconstruct the database because the replication may have been started with

```
var db = new PouchDB('...');
var remote = new PouchDB('....');
db.sync(remote);
```

I think this is mostly a bad decision we made early on with construction + taskqueue which we have seem come up a lot with errors during construction etc.

I think we want to move to a stateless constructor that can never fail, it means all of our error handling goes via a promise api, we can get rid of taskqueue and the various bugs it causes. In Firefox OS we generally do this with something like

```
function Constructor(opts) { 
  this.opts = opts;
}

function openDb (opts) { 
  if (dbs[opts.name]) { 
    return Promise.resolve(dbs[opts.name]);
  } 
  // else do construction etc
}

function post(doc) { 
  openDb(this.opts).then(function(idb) { 
     idb.createTxnWhatever
  });
}
```

**Comment by triniwiz** [comment]
Anything i could do till there's a fix  ??

**Comment by daleharvey** [comment]
`skipSetup` technically should do the same as my proposal in #4057, but in my testing it didnt so a little confused

**Comment by nolanlawson** [comment]
fixed by f8c06ac76d321e84bdcab25f9cf02f5d648e06af

**Comment by baversjo** [comment]
thanks for your hard work!
